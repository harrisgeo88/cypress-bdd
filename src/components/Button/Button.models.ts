import { ReactChild, HTMLProps } from 'react'

export interface ButtonProps extends HTMLProps<HTMLButtonElement> {
  onClick: () => void
  children: ReactChild
}
