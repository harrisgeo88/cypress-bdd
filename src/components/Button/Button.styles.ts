import styled from 'styled-components'

export const ButtonWrapper = styled.button.attrs({ type: 'button' })`
  margin: 20px 0;
  height: 50px;
  width: 100%;
  background-color: green;
  color: white;
  border-radius: 5px;
`
