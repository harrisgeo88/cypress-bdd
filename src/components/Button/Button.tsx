import React from 'react'
import { ButtonWrapper } from './Button.styles'
import { ButtonProps } from './Button.models'

export const Button = ({ children, ...rest }: ButtonProps) => {
  return <ButtonWrapper {...rest}>{children}</ButtonWrapper>
}
