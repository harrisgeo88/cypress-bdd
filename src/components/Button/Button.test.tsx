import { Button } from './Button'
import { render } from '@testing-library/react'

describe('given an `Button` component', () => {
  describe('when it is rendered', () => {
    it('should match snapshot', () => {
      const { container } = render(<Button onClick={jest.fn()}>Go</Button>)
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
