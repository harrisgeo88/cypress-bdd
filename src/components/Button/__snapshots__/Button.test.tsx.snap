// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given an \`Button\` component when it is rendered should match snapshot 1`] = `
<button
  class="Buttonstyles__ButtonWrapper-sc-11g5hpj-0 gvIbjm"
  type="button"
>
  Go
</button>
`;
